Webapp/sources/services/data_export/eln_file.py

```diff
@@ -0,0 +1,555 @@
+
+    def _define_components(self):
+        """Defines the components being described in the metadata json"""
+        self._define_authors()
+        for reaction in self.data_export_request.reactions:
+            export_rxn = ELNExportReaction(reaction, self)
+            export_rxn.make_export_files()
+            export_rxn.get_files()
+            export_rxn.get_parts()
+            self.reaction_list.append(export_rxn)
+            # get the root parts
+            self.root_parts.append(export_rxn.defined_dataset)
+
+    def _make_ro_crate_metadata_json(self):
+        """Makes the metadata json file that describes the export contents."""
+        ro_crate_graph = []
+        ro_crate_graph += get_constant_ro_crate_start()
+        # update the ro-crate with export specific data
+        ro_crate_graph.append(self._describe_root_directory())
+
+        for rxn in self.reaction_list:
+            ro_crate_graph.append(rxn.defined_dataset)
+
+        for rxn in self.reaction_list:
+            [ro_crate_graph.append(file) for file in rxn.defined_files]
+
+        for rxn in self.reaction_list:
+            [ro_crate_graph.append(comment) for comment in rxn.defined_comments]
+
+        ro_crate_graph += [author for author in self.defined_authors]
+
+        # define the contents of the metadata json
+        ro_crate_metadata_contents = {
+            "@context": "https://w3id.org/ro/crate/1.1/context",
+            "@graph":
+            # the graph is a list of dictionaries defining the elements and their structure
+            ro_crate_graph,
+        }
+        # save as a file to memory then upload the blob to azure.
+        self.memory_file = io.StringIO()
+
+        with self.memory_file as f:
+            json.dump(ro_crate_metadata_contents, f)
+            self.file_contents = bytearray(f.getvalue(), "utf-8")
+            self.content_size = f.seek(0, os.SEEK_END)
+
+        sources.services.data_export.export.save_blob(
+            self.container_name,
+            self.root + "/" + "ro-crate-metadata.json",
+            self.file_contents,
+        )
+
+    def _describe_root_directory(self):
+        """Describes the root directory by listing the reaction IDs in hasPart for the metadata json"""
+        return {
+            "@id": "./",
+            "@type": "Dataset",
+            "hasPart": [{"@id": part["@id"]} for part in self.root_parts],
+        }
+
+    def _define_authors(self):
+        """List of all authors of the exported reactions"""
+        [
+            self.defined_authors.append(self._get_author(reaction))
+            for reaction in self.data_export_request.reactions
+        ]
+        # remove duplicates
+        self.defined_authors = [
+            i
+            for n, i in enumerate(self.defined_authors)
+            if i not in self.defined_authors[:n]
+        ]
+
+    @staticmethod
+    def _get_author(reaction: models.Reaction) -> Dict:
```

Comment:
Why do you have a static method that is private? If this is only used inside the class, you can remove the `@staticmethod` decorator and add `self` as the first argument to the method

Reply:
Originally made it static because the function didn't reference self, but I can do it as you described going forwards. Also, there are a few more instances of this from the [previous PR](https://github.com/AI4Green/AI4Green/blob/main/Webapp/sources/services/data_export/export.py) eg., line 222. I can change these too for consistency.

Reply:
Actually, I'm wondering now if it would come up as a warning in your IDE saying that you can make it static because `self` is not referenced. I was thinking that using static was when you might have something public that doesn't require an instance of the class. But know you mention it, it does make sense that it can be static for the reason you said. You can leave this and the other instances, I think.